corehq/apps/hqwebapp/middleware.py

```diff
@@ -0,0 +1,22 @@
+from django.middleware.csrf import CsrfViewMiddleware, REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN
+
+from corehq.util.soft_assert import soft_assert
+from django.conf import settings
+
+
+class HQCsrfViewMiddleWare(CsrfViewMiddleware):
+
+    def _reject(self, request, reason):
+        if settings.CSRF_ALWAYS_OFF and reason in [REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN]:
+
+            warning = "Request at {url} doesn't contain a csrf token. Letting the request pass through for now. "\
+                      "Check if we are sending csrf_token in the corresponding POST form, if not fix it. "\
+                      "Read more here https://github.com/dimagi/commcare-hq/pull/9227".format(
+                          url=request.path
+                      )
+            _assert = soft_assert(notify_admins=True, exponential_backoff=True)
```

Comment:
My guess is that it will be too spammy. I might recommend two changes:

1. Only trigger for some fraction of requests (can just use `random() < some_probabilty` to decide whether to send)
2. Only send to yourself while you're actively working on this.

I also think we should deploy to staging for several days to catch as many as possible before merging. (those could also be done in independent PRs with `@csrf_protect` which will likely make conflicts easier)